lib/entities/QuoteResponse.ts

```diff
@@ -67,31 +69,25 @@ export class QuoteResponse implements QuoteResponseData {
     );
   }
 
-  public static fromRFQ(args: FromRfqArgs): ValidatedResponse {
+  public static async fromRFQ(args: FromRfqArgs, provider?: ethers.providers.JsonRpcProvider, log?: Logger): Promise<ValidatedResponse> {
```

Comment:
this is a bit of a meta point but would prefer to not have business logic within these entity classes, it may be cleaner to separate out the _creation_ of a quote request/response from whether it is valid or not from dependencies on external services. That would allow us to not have to have an optional provider field here

Reply:
LMK if this is what you had in mind:
https://github.com/Uniswap/uniswapx-parameterization-api/pull/399/commits/d0c19ddf2a4d99f7fcfa70e99ac3b4a3c3b7f7bf#diff-7a29254ba6f168fefbeb5305206da26cda2f763b9d76a415c32f17a6d5498353R166